Add 'hide dust payments' preference toggle

leofelix077 opened the pull request:
Closes https://github.com/stellar/freighter-mobile/issues/247 


Tests done with asset_type 'credit_alphanum4' instead of 'native'



https://github.com/user-attachments/assets/ee498fa2-8962-4996-8a20-5aa65f117beb

CassioMG commented on src/hooks/useGetHistoryData.ts:
```diff
@@ -113,9 +133,10 @@ function useGetHistoryData(
 
       const balances = getBalances();
 
+      setRawHistory(history);
```
I see we are calling `setRawHistory` here and then `setHistoryData` on line 142 and then `setHistoryData` will also be called inside useEffect on line 106 with basically the same inputs. Why is that?

danilosilvackl commented on src/hooks/useGetHistoryData.ts:
```diff
@@ -89,10 +92,27 @@ function useGetHistoryData(
+    Horizon.ServerApi.OperationRecord[]
+  >([]);
+
+  useEffect(() => {
+    if (rawHistory.length) {
+      const balances = getBalances();
+      setHistoryData({
+        balances,
+        history: createHistorySections(
+          publicKey,
+          rawHistory,
+          isHideDustEnabled,
+        ),
+      });
+    }
+  }, [isHideDustEnabled, publicKey, rawHistory, getBalances]);
```
if i understood this correctly, this is responsible to update the history in case the the settings change right?
What if we leave modify the `fetchData` to receive the isHideDustEnabled as a parameter and leave the HistoryScreen responsible to send this information? Im considering this to avoid running code in the background without a need, so the fetch function would only change when the user loads the page again, does that makes sense?

CassioMG commented:
> Closes #247
> 
> Tests done with asset_type 'credit_alphanum4' instead of 'native'
> 
>  Screen.Recording.2025-08-08.at.11.56.56.mov

it took a while for me to realize it, but I think on your video the payments are being hidden because you are swapping a small value of XLM for SSLX. Is that right? And not because we have small values of SSLX which is actually the desired behavior since we are only interested in XLM here so no changes needed here. I'm highlighting it here just in case someone else has the same question because the History rows do not display the XLM amounts, only the SSLX amounts.

leofelix077 commented:
@CassioMG  yes. I mentioned on the description about the flip. At the moment i don't have residual payments in my account, so I switched the asset type in code for testing and reverted before committing

CassioMG commented:
> @CassioMG yes. I mentioned on the description about the flip. At the moment i don't have residual payments in my account, so I switched the asset type in code for testing and reverted before committing

Ah ok, now I understand what you mentioned by the description
